### ISSUE TITLE
Freeform not sending user and admin emails (Sendgrid)

### ISSUE THREAD

**Issue by ryanmasuga** [solspace/craft3-freeform#269]
**Describe the bug or issue you're experiencing**
We are using Sendgrid as transactional email service to send mail from our site. In the email settings we can click "Test" and successfully receive email via Sendgrid. We've discovered that from Freeform forms on the front end of our site, no emails are being sent (though form submission are logged.) 

We see this error in the logs: 

```
Error sending email: Expected response code 250 but got code "550", with message "550 Unauthenticated senders not allowed
```

This doesn't seem to make sense if we can test the email settings from the control panel with success.

**Steps to reproduce**

1. Submit Freeform form (Contact, info...any form on our site)
2. See submission under Freeform "Submissions", but receive no emails. 

**Expected behavior**
Admin should get a notification email every time, and on certain forms, the submitter should get an email.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.26
- Freeform Version: 3.2.11
- Freeform Edition: Pro
- Fresh Install or Upgrade: Longstanding site

**Additional context**
We could find this info on Sendgrid's site: https://docs.sendgrid.com/for-developers/sending-email/smtp-errors-and-troubleshooting#550-unauthenticated-senders-not-allowed

**Comment by kjmartens** [comment]
Hi @ryanmasuga, 

Does the value of the "From Name" in the email notification template match the email address configured for Sendgrid/inside the Craft Email settings area? I believe it needs to be in this case. If it doesn't match, it would also explain why it works for the test inside the Email area but not in Freeform. 🙂 

If you wish to use a different email address for the "from" email, we would encourage you to use the **Reply-to Email** (and Reply-to Name) fields instead.

https://docs.solspace.com/craft/freeform/v3/overview/email-notifications.html#template-options

**Comment by ryanmasuga** [comment]
The Notification “From” settings do match the site’s email settings. (Both are hello@)

**Comment by kjmartens** [comment]
Ok, thanks for clarifying. Freeform just uses Craft's API to send emails - not anything separate or special on our end. Can you see if anything else in Craft sends email notifications?  If you have Craft Pro edition, you can use the "Send password reset email" setting on another user account:

<img width="480" alt="Screen Shot 2022-02-08 at 10 16 10 AM" src="https://user-images.githubusercontent.com/1821674/153029063-d93512a9-133f-45e8-9581-19d59a2423c5.png">

**Comment by ryanmasuga** [comment]
I created a new API key at Sendgrid to "start from scratch" and that seemed to fix this. The internet is fun! Thanks for your time.

**Comment by kjmartens** [comment]
Happy to help! Thanks for letting us know, and hope you have a great evening. 🙂

**Comment by AbbeyDesign** [comment]
I am getting this also but a slightly different flavor(530 instead of 550) and different transactional service(Sparkpost). Test emails from the control panel send but admin email notifications from Freeform do not.

From the logs
Error sending email: Expected response code 250 but got code "530", with message "530 5.7.1 Authorization required. Ref. https://developers.sparkpost.com/api/index#header-smtp-relay-endpoint

**Comment by kjmartens** [comment]
@AbbeyDesign Is it possible that your password/login information for Sparkpost has changed recently? I'm not suggesting you didn't update your credentials in the Craft Email settings (it sounds like you must have), but I am curious if this is the case nonetheless. 🙂

**Comment by AbbeyDesign** [comment]
@kjmartens Nothing changed via Sparkpost. 


I was able to resolve this by going into the CP and resaving Craft's email settings. Not just resaving, but changing one or two settings, saving those settings, and then re-enter the original settings and resaving. This happened recently where Craft's S3 plugin credentials settings stopped working. We had to go into the plugin settings and not just resave, but change a field, save, and then put the original info back in and save again.

So one could assume this was an issue with Craft, but emails from the cp worked, it was only Freeform that had the issue.
